Added segment for Open Weather Map.

Descalon opened the pull request:
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

Added a new segment that pulls data from [Open Weather Map](https://openweathermap.org) to display current weather and temperature.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

Descalon committed:
feat(segment): added segment for Open Weather Map

JanDeDobbeleer commented:
I'm currently on holiday so I'll be reviewing this a bit slower than usual.

Descalon committed:
fix(docs): fixed markdown linting issues

JanDeDobbeleer commented on docs/docs/segment-owm.md:
```diff
@@ -0,0 +1,38 @@
+
+Shows the current weather of a given location.
+
+**NOTE**: You **must** request an API key at the [Open Weather Map](https://openweathermap.org/price) website.
```
You can use the [caution admonition](https://docusaurus.io/docs/markdown-features/admonitions) for this.

Descalon replied:
Added in 3bdacea

JanDeDobbeleer commented on docs/sidebars.js:
```diff
@@ -56,6 +56,7 @@ module.exports = {
         "text",
         "time",
         "ytm",
+        "owm"
```
Can you add this in alphabetical order?

Descalon replied:
Modified in e31bb54

JanDeDobbeleer commented on src/segment_owd.go:
```diff
@@ -0,0 +1,88 @@
+	icon := ""
+	switch q.Data[0].TypeID {
+	case "01d":
+		icon = "滛"
```
Can you add the string based representation of the icons? We always use `\uxxxx` for this. This will also make it so they aren't adjustable, could run into issues based on the font used. For our `os` segment we exposed all platform icons for example.

Descalon replied:
Updated in 538ab69

JanDeDobbeleer commented on src/segment_owd_test.go:
```diff
@@ -0,0 +1,63 @@
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOWDStringPlayingSong(t *testing.T) {
```
I don't think the function name is correct

Descalon replied:
That's a whoopsie 😅. Removed in c547971

JanDeDobbeleer commented on src/segment_owd_test.go:
```diff
@@ -0,0 +1,63 @@
+
+	assert.Equal(t, expected, d.string())
+}
+func bootstrapOWDTest(json string, err error) *owm {
```
We moved to another style of tests to facilitate adding use cases. See [here](https://github.com/JanDeDobbeleer/oh-my-posh/blob/572149b176b864775edae0e531ed38153a12d66a/src/segment_battery_test.go#L17) for example.

Descalon replied:
Updated in c547971

Descalon committed:
chore: added caution admonition

Descalon committed:
chore: reorderd segments to be alphabetical

Descalon committed:
chore: changed font icons over to hexcodes

Descalon committed:
test: changed OWD tests to a case style

Per pull request comment, changed the owd tests to resemble
the tests for the Battery segment

Descalon committed:
Merge branch 'JanDeDobbeleer:main' into main

Descalon committed:
Merge branch 'main' of github.com:Descalon/oh-my-posh

lnu commented:
I tried the segment but I get this in the  code:
![image](https://user-images.githubusercontent.com/1829553/129681215-c3872460-1beb-42a0-9413-1bda6e08738a.png)

but the url works in the browser:
![image](https://user-images.githubusercontent.com/1829553/129681260-d7ec0706-b0ed-4f46-876c-d01759695735.png)

JanDeDobbeleer commented:
@lnu how long does the request take? There's a timeout set at 20ms for get requests in `environment` if I'm not mistaken.

JanDeDobbeleer commented:
@all-contributors please add @Descalon for code

lnu commented:
@JanDeDobbeleer  it takes between 20 and 30 ms with postman. So yes I had to raise the timeout.
